metaphlan/3.0.3/Dockerfile

```diff
@@ -0,0 +1,60 @@
+FROM ubuntu:bionic AS builder
+
+LABEL base.image="ubuntu:bionic"
+LABEL version="1"
+LABEL software="MetaPhlAn3.0"
+LABEL software.version="3.0"
+LABEL description="microbial composition of metagenomes"
+LABEL website="http://cab.spbu.ru/files/release3.13.0/manual.html"
+LABEL maintainer="Tara Gallagher"
+LABEL maintainer.email="tgallagher@utah.gov"
+
+# install python (>3.6) and other dependencies
+RUN apt-get update && apt-get install -y software-properties-common && \
+	add-apt-repository ppa:deadsnakes/ppa&& \
+	apt-get update && apt-get install -y \
+        gcc \
+	wget \
+	python3.7 \
+	python3.7-dev \
+	python3-distutils \
+	python3-setuptools \
+        python3-pip \
+	unzip && \ 
+	python3.7 -m pip install pip --force-reinstall && \
+	python3.7 -m pip install numpy Cython six --force-reinstall && \
+	ln -s /usr/bin/python3.7 /usr/bin/python 
+
+
+# bowtie 2 dependency 
+RUN mkdir /usr/bin/bowtie2 && \
+	cd /usr/bin/bowtie2 && \  
+	 wget https://sourceforge.net/projects/bowtie-bio/files/bowtie2/2.3.3.1/bowtie2-2.3.3.1-linux-x86_64.zip/download && \
+	unzip download && \
+	rm download 
+
+ENV PATH="$PATH:/usr/bin/bowtie2/bowtie2-2.3.3.1-linux-x86_64"
+
+
+# install metaphlan 3
+RUN python3.7 -m pip install metaphlan==3.0.3
+
+# download metaphlan database
+RUN metaphlan --install && \ 
+	mkdir /data 
+
+# build onto first stage
+FROM ubuntu:bionic
+
+COPY --from=builder /usr/local/bin /usr/local/bin
+COPY --from=builder /usr/bin/bowtie2/ /usr/bin/bowtie2/
+COPY --from=builder /usr/bin/python3.7 /usr/bin/python3.7
+COPY --from=builder /usr/lib/python3.7 /usr/lib/python3.7
+COPY --from=builder /usr/lib/x86_64-linux-gnu/libexpat.so /usr/lib/x86_64-linux-gnu/libexpat.so
+COPY --from=builder /lib/x86_64-linux-gnu/ /lib/x86_64-linux-gnu/
+COPY --from=builder /usr/local/lib/python3.7/dist-packages/ /usr/local/lib/python3.7/dist-packages/
+COPY --from=builder /usr/lib/python3/dist-packages/* /usr/lib/python3.7/dist-packages/
+
+WORKDIR /data
+
+ENV PATH="$PATH:/usr/bin/bowtie2/bowtie2-2.3.3.1-linux-x86_64"
```

Comment:
Please edit/add this line to ensure compatibility with singularity:
```Dockerfile
ENV PATH="$PATH:/usr/bin/bowtie2/bowtie2-2.3.3.1-linux-x86_64"\
    LC_ALL=C
```
Probably unnecessary, but maybe some perl needs to run somewhere that isn't obvious

Reply:
Thanks, I added the env path to dockerfiles.

For the perl comment, I'm assuming this is just a simple check to make sure locale settings are in the image. I added an empty print line (RUN perl -e 'print') in each dockerfile (for 3.0.3 = line 87 and 3.0.3-no-db = line 91). Is this what you meant? Let me know if you want me to change

Reply:
Yeah, for some reason when docker images are converted to singularity format, locale settings are not saved and perl programs run into locale settings errors. Doesn't happen with docker, only singularity. That's why I typically will set LC_ALL=C to make sure its baked into the image

RE lines 87 and 91 in dockerfiles - I didn't realize you could check it that way, that helps! Only downside is that for the no-db dockerfile, line 91 adds an additional layer if you give it its own RUN statement. Perhaps combine `perl -e 'print'` test with the previous layer?
```Dockerfile
# make dir for metaphlan database
RUN mkdir /usr/local/lib/python3.7/dist-packages/metaphlan/metaphlan_databases && \
 ln -s /usr/bin/python3.7 /usr/bin/python && \
 perl -e 'print'
```

Reply:
Actually, I'm not sure if perl will always generate a warning if it's unable to find locale settings. I need to look further into this. Since this is still new to me 😅 , I'll change to whatever you typically do in regards to your 1st comment - "Probably unnecessary, but maybe some perl needs to run somewhere that isn't obvious". Do you have an example in one of your dockerfiles? I'll also fix the no-db dockerfile and get rid of running that extra layer. 

thanks!

Reply:
it's an annoying ~~error~~ warning that looks like this. I pulled an older, non-staphb prokka docker image (https://hub.docker.com/r/hadrieng/prokka/dockerfile) with singularity, entered singularity shell, and pulled up help options.
```
Singularity> prokka --help
perl: warning: Setting locale failed.
perl: warning: Please check that your locale settings:
        LANGUAGE = (unset),
        LC_ALL = (unset),
        LANG = "en_US.UTF-8"
    are supported and installed on your system.
perl: warning: Falling back to the standard locale ("C").
Name:
  Prokka 1.12 by Torsten Seemann <torsten.seemann@gmail.com>
```